librz/util/buf.c

```diff
@@ -359,14 +426,34 @@ RZ_API RzBuffer *rz_buf_new_sparse_overlay(RzBuffer *b, RzBufferSparseWriteMode
 	return new_buffer(RZ_BUFFER_SPARSE, &cfg);
 }
 
-RZ_API RzBuffer *rz_buf_new_with_buf(RzBuffer *b) {
+/**
+ * \brief Creates a new buffer from a source buffer.
+ * \param b The source buffer used to create the sub buffer.
+ * \return Return the new allocated buffer.
+ *
+ * \see rz_buf_new_with_bytes()
+ *
+ * The function creates a new allocated buffer using the RZ_BUFFER_BYTES back end.
+ * WARNING: this function doesn't use reference counting, this can lead to a lot
+ * of memory overhead.
+ */
+RZ_API RZ_OWN RzBuffer *rz_buf_new_with_buf(RzBuffer *b) {
 	ut64 size = 0;
 	const ut8 *tmp = get_whole_buf(b, &size);
 
 	return rz_buf_new_with_bytes(tmp, size);
 }
 
-RZ_API RzBuffer *rz_buf_new_with_bytes(RZ_NULLABLE const ut8 *bytes, ut64 len) {
+/**
+ * \brief Creates a new buffer with a bytes array.
+ * \param bytes The bytes array used to initialized the buffer.
+ * \param len The length of the bytes array.
+ * \return Return the new allocated buffer.
+ *
+ * The function creates a new allocated buffer using the RZ_BUFFER_BYTES back end.
+ * The bytes parameter can be NULL, but the length should be set to 0.
```

Comment:
The function creates a new buffer in memory, initializing it with the bytes passed as argument. The bytes parameter can be NULL, but the length should be set to 0.